feat(integrations): 1st Iteration Frontend

vpandiarajan20 opened the pull request:
Made a page to create SentryFunctions

vpandiarajan20 committed:
feat(integrations): 1st Iteration Frontend

Made a page to create SentryFunctions

scefali commented on static/app/routes.tsx:
```diff
@@ -797,6 +797,16 @@ function buildRoutes() {
         />
+        <Route
+          path="new-sentry-function/"
+          name={t('Create Integration')}
```
@vpandiarajan20 can you update the name?

scefali commented on static/app/routes.tsx:
```diff
@@ -797,6 +797,16 @@ function buildRoutes() {
           )}
         />
+        <Route
+          path="new-sentry-function/"
```
@vpandiarajan20 We should have sentry functions as their own directory like: `https://sentry.io/settings/:orgSlug/developer-settings/sentry-functions/new/`

The reason we should do this is so for the edit Sentry functions view, we can have it at `https://sentry.io/settings/:orgSlug/developer-settings/sentry-functions/:functionSlug/`. If we had it at `https://sentry.io/settings/:orgSlug/developer-settings/functionSlug/`, then we wouldn't know if it was a Sentry function or public/internal integration.

scefali commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,62 @@
+class SentryFunctionFormModel extends FormModel {
+  getTransformedData(): {} {
+    const data = super.getTransformedData() as Record<string, any>;
+    const {...output} = data;
```
@vpandiarajan20 why do you have this line?

scefali commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,62 @@
+        apiMethod={method}
+        apiEndpoint={endpoint}
+        model={form}
+        onPreSubmit={handlePreSubmit}
```
@vpandiarajan20 You can do this with an anonymous function to make this cleaner:
```
onPreSubmit={() => addLoadingMessage(t('Saving changes..'))}
```

joshuarli committed:
[skip ci] gitbot test

joshuarli committed:
[skip ci] gitbot test

vpandiarajan20 committed:
style(integrations): route family fix & remove redundant code

replaced function call with anonymous function
created family for sentry function routes, to prevent confusion with integrations

joshuarli committed:
[skip ci] gitbot test 2

joshuarli committed:
[skip ci] gitbot test 2

joshuarli committed:
[skip ci] gitbot test 3

joshuarli committed:
[skip ci] gitbot test 4

joshuarli committed:
[skip ci] gitbot test 5

asottile-sentry committed:
ref: convert legacy compat.map serialize calls to listcomps (#36532)

JonasBa committed:
fix(selectedEnvs): fix eslint deps (#36577)

* fix(selectedEnvs): fix deps

* fix(environmentpagefilters): fix typo

leedongwei committed:
feat(team-roles) Add Team Roles to organizationTeams (#35667)

leedongwei committed:
ref(team-roles) Rename methods for org-level roles (#36497)

leedongwei committed:
ref(team-roles) Move OrganizationTeamMemberDetailsEndpoint (#36549)

danecando committed:
feat(replays): adds compact mode for player control bar (#36483)

Changes
- Adds resize observer to player control bar
- Adds compactLevel state for possible multi breakpoints
- Updates player control UI when compact is set

Note
We might not actually need breakpoints. I think it's good to go as is as we have and should have a min width for the sidebar

Closes #36272

k-fish committed:
ref(sdk): Bump python to sdk 1.7 (#36575)

Bump the sdk again.

Refs #36502
Refs #36539

Co-authored-by: Neel Shah <neelshah.sa@gmail.com>

dashed committed:
fix(hybrid-cloud): Fix org slugify on saving new instances of Organization (#36604)

narsaynorath committed:
feat(dashboard-filters): Update endpoint to save filters (#36462)

Exposes project, environment, start, end, range, and filters as keys
in the dashboard detail API. Only releases is supported as a field in
filters.

Commit:
ref(projectconfig): Do not sample metrics for v2 (#36612)

There are not many of these, sampling might distort results.

sigmaris committed:
fix: replace usage of global functions in Lua with locals (#34416)

matejminar committed:
feat(sampling): Remove transaction and user id condition (#36613)

ahmedetefy committed:
feat(dyn-sampling): Add BE validation for uniform rules [TET-191] (#36607)

* feat(dyn-sampling): Add BE validation for uniform rules [TET-191]

Adds validation that ensures uniform sampling rule
cannot be deleted, no multiple uniform rules can
co-exist and unifrom rule is always in the last position

Commit:
fix(projectconfig): Give the build task debounce key a short TTL (#36514)

The TTL of the debounce key is made to match the hard timeout of the
celery task.  This ensures that a debounce key can not live longer
than the task can, and thus no sticky debounce key occurs when a task
times out.  Because the schedule time is ignored this could result in
unnecessary duplicate tasks, however if a task already took this long
it probably wasn't going to succeed anyway.

Secondly this sets a global expiry time for the task itself, an
expired task will stay in RabbitMQ but the Celery worker will ack it
without running the task.  This ensures that if workers are shut down
during incidents or similar previous tasks will not be consuming
computation resources.

INGEST-1441
INGEST-1442

priscilawebdev committed:
ref(sampling): Add more analytic code for sampling - [TET-173] (#36513)

JonasBa committed:
feat(profiling): split table from renderer (#36471)

* feat(profiling): split table from renderer

* ref(flamegraph): noopColor -> noopDuration

* fix(flamegraph): add helper import

* feat(profiling): always display frame table (#36482)

* feat(profiling): always display frame table

* feat(profiling): allow table to be positioned to left, bottom or right side of the chart. (#36523)

* feat(profiling): add layout to preferences

* feat(profiling): position table left and right

* fix(flamegraph): root

* ref(profiling): improve terminology (#36593)

* ref(framestack): rename tableRoot to referenceRoot

* ref(framestack): rename roots

* ref(framestack): rename roots in test

* ref(framestack): final rename

* feat(profiling): use new layout icons (#36596)

k-fish committed:
fix(perf): Fix asset measurements sending NaN/null (#36609)

The nullish coalescing wasn't wrapped in parens so it was ambiguous. This should fix this specific case happening again, but we should also look into why the transfer sizes aren't even defined, which was unexpected.

k-fish committed:
feat(ui): Add 'ui' as a span breakdown (#36515)

* feat(ui): Add 'ui' as a span breakdown

This will add 'ui' as another span breakdown in the transaction summary. It will help show the percentage time spent in UI for mobile and frontend projects.

markstory commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,57 @@
+}
+
+export default function sentryFunctionDetails(props: Props) {
+  const form = new SentryFunctionFormModel();
```
You'll need a `useRef` for this model. Right now each render of this component will make a new form model and erase the current values. If you used `useRef`, instead you'd get the same form model for all render passes.

markstory commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,57 @@
+];
+
+class SentryFunctionFormModel extends FormModel {
+  getTransformedData(): {} {
+    return super.getTransformedData() as Record<string, any>;
+  }
+}
```
Unless you have custom logic you're adding here this class could be removed.

vpandiarajan20 replied:
Agreed, will add again later when needed.

matejminar committed:
feat(sampling): Format and round sample rates [TET-211] (#36617)

armenzg committed:
ref(ci): Bump timeout of acceptance jobs (#36618)

I've hit this more than once with one of the acceptance jobs, thus, needing to re-run in order to trigger Visual Snapshots.

markstory committed:
fix(api) Fix a 500 when updating project rule history (#36590)

Fixes SENTRY-SHD

edwardgou-sentry committed:
fix(discover): events endpoint default rate limit if missing org and cant check feature flag (#36614)

Return the default rate limit if we cant get the org. Eventually we will remove the feature flag and always return the custom events rate limit so this edge case can no longer happen.

Zylphrex committed:
tests(discover): Improve stability of events geo tests (#36619)

When using a timestamp 1 minute in the past for test events, the end timestamp
on the query can sometimes be equal due to the timestamp fuzzing logic. Since we
use a less than (<) condition for the end timestamp, the query will return no
result in this situation leading to a flakey test. This aims to improve the
stability of the test by storing the event further in the past so it won't
collide with the end timestamp on the query.

mitsuhiko committed:
feat(db): Add support for JSON in pickle field (#36350)

doralchan committed:
added contract expand and rewind icons (#36598)

matejminar committed:
feat(sampling): Handle empty current series (#36620)

danecando committed:
feat(replays): Sidebar Left Layout (#36429)

Adds an additional option for the layout param that positions that sidebar on the left side

Use `?l_page=sidebar_left` in the url to enable the new arrangement.

Fixes #36410

danecando committed:
feat(replays): Updated player control tooltips and icons (#36516)

Updates tooltip names and aria-labels for replay player controls & makes use of new `Rewind10`, `Expand`, and `Contract` icons in the player controls.

Closes #36527
Closes #36477

armenzg committed:
fix(ui): Sentry Apps: Vertical CircularIndicator and capitalize (#36576)

For instance, when loading the Linear app, the circular indicator would not line up with the text.

You can see this working by running `yarn dev-ui` (sign in with a non-SSO org) and load this page:
https://localhost:7999/settings/sentry-ecosystem/sentry-apps/linear/

This also capitalizes the name of the app when the installing/uninstalling apps.

Fixes Jira ticket: API-1173

wedamija committed:
feat(mep): Remove `entity_key` and `time_col` from `BaseEntitySubscription` (#36460)

This removes `entity_key` and the derived `time_col` from the entity subscription classes. I'm doing
this so that we can have an entity subscription that will be able to produce a query for any of the
metric entities (count, sets, etc) without having to manually define it. This allows us to use
`QueryBuilder` to determine the entity instead of having to parse the aggregates ourselves.

We can then use `QueryBuilder` to fetch the entities in place of these previously hardcoded values.

malwilley committed:
fix(ui): Clear TimeRangeSelector input value on select and close (#36595)

wedamija committed:
feat(mep): Move `type` to `SnubaQuery` (#36490)

It turns out that we need `type` at a lower level than the alert. This allows us to determine which
entity a `SnubaQuery` is associated with, since we need to know both the type and dataset to
determine it.

vuluongj20 committed:
fix(alert): Don't collapse alert when clicking on expanded content (#36592)

* fix(alert): Don't collapse alert when clicking on expanded content

* fix(alert): Fix test

* ref(alert): Fix internal paddings

Commit:
feat(msteams): Add default installation_type for new integrations (#36534)

evanpurkhiser committed:
ref(ci): search-syntax fixture changes trigger frontend + backend (#36606)

wedamija committed:
feat(mep): Backfill and mark `SnubaQuery.type` as not null (#36493)

Modified version of the backfill from `AlertRule.type`. Test is commented out because of a bug with
migration tests when rolling back not null columns.

evanpurkhiser committed:
ref(storybook): Use Hovercard for storybook icons (#36430)

scttcper committed:
test(ui): Convert admin views to RTL (#36628)

jesus4497 committed:
feat(replays): Iterate on Breadcrumb styles  (#36396)

Added new styles to the breadcrumb component which highlights in a better way when the user is on its event.

Closes #36112

vuluongj20 committed:
fix(selectAllNotice): Use Alert component, not alertStyles (#36627)

scttcper committed:
fix(workflow): Remove issue-alerts-targeting feature backend (#36164)

Commit:
feat(migration): MsTeams backfill default installation type (#36536)

asottile-sentry committed:
ref: don't use deprecated yield_fixture (#36633)

markstory committed:
ref(tests) Try disabling internal error collection (#35596)

* ref(tests) Try disabling internal error collection

I've noticed that when tests have problems our test output gets
cluttered with timeouts from the python SDK trying to submit errors back
to a host that doesn't exist or isn't responding.

My hope is that by disabling the internal project key we can skip all of
that and have faster, less noisy tests.

* Fix tests.

* See if mutating settings will fix these tests in CI

* Close docker clients after using them

I was getting a few unclosed stream errors locally, and closing
docker clients has helped resolve that.

wedamija committed:
feat(mep): Restructure how we determine entity subscription for alerts (#36605)

Previously we mapped a specific `EntityKey` to all `EntitySubscription` classes. As part of
introducing metric based performance alerts, we want to have the `EntitySubscription` determine the
specific entity that the subscription will run on. This allows us to automatically determine the
correct entity for metric based alerts without having to duplicate logic that parses
aggregates/datasets/etc.

asottile-sentry committed:
ref: fix pip/wheel flipflop by using the pinned versions (#36636)

asottile-sentry committed:
ref: add pytest-xdist to development requirements (#36630)

asottile-sentry committed:
ref: replace legacy compat.map with list comprehensions in integrations (#36642)

dashed committed:
feat(hybrid-cloud): Adjust CORS for customer domains (#36639)

asottile-sentry committed:
ref: replace legacy compat.map with list comprehensions in similarity (#36643)

asottile-sentry committed:
Revert "ref: fix pip/wheel flipflop by using the pinned versions (#36636)" (#36648)

This reverts commit 6d08d3064c4a29b9175d238ea0198e056523009d.

taylangocmen committed:
feat(active-release): Active Release Notification Email redesign (#36425)

* add option for passed initials to get_letter_avatar

* add platform to project in debug mail

* everything else

* filter context keys

* remove id

* add changes to email

* add users_seen to context

* fix test

* fix test

* update text email

* fix test

priscilawebdev committed:
fix(ui): Fix expand and collapse button not working - issue error warning (#36616)

joshuarli committed:
test: upgrade responses to latest (#36597)

edwardgou-sentry committed:
feat(dashboard): dashboard widget builder only allow cm on supported functions (#36570)

updates widget builder yaxis and column function arg dropdown to only show custom measurements when supported by the function

untitaker committed:
fix(test): Use correct url prefix for minidump test (#36635)

wmak committed:
feat(mep): Generic metric dataset (#36623)

- This properly switches the metric query builder over to the generic metrics dataset
- This currently means that local metrics data will no longer work
- This makes a bit of a TODO to come back and rename the
  Dataset.Metrics to Dataset.PerformanceMetrics
- Also defaulting to Metrics for now as the default dataset to get tests
  passing, but will likely make it PerformanceMetrics instead in the
  long run
- This also changes the hack to only happen when on the
  PerformanceMetrics dataset

Zylphrex committed:
tests(discover): Improve stability of eventsv2 tests (#36641)

Same motivation as #36619, this aims to improve the stability of the eventsv2
tests by moving the event timestamps further in the past.

scttcper committed:
test(ui): Convert issue sync list to RTL (#36656)

scttcper committed:
test(ui): Convert issue diff to RTL (#36657)

scttcper committed:
test(ui): Convert ignore to RTL (#36662)

scttcper committed:
test(ui): Convert global modal to RTL (#36659)

scttcper committed:
test(ui): Convert stream group to RTL (#36654)

scttcper committed:
test(ui): Remove mixed emotion modal and rtl modal (#36658)

scttcper committed:
test(ui): Convert feature tour modal RTL (#36660)

scttcper committed:
test(ui): Convert repository row to RTL (#36655)

scttcper committed:
test(ui): Convert toggle raw event to RTL (#36652)

Zylphrex committed:
fix(trace-view): Left pad root span id (#36671)

The root span id is stored as an UInt64 in ClickHouse. And when querying, it is
converted to a hex string using a naive base 10 to base 16 conversion. This
conversion does not take into account that some smaller values need to be left
padded with 0s in order to have a length of 16. So sometimes, the parent span
ids returned from snuba does not match the span ids retrieved from nodestore.

scttcper committed:
test(ui): Convert platform picker to RTL (#36640)

edwardgou-sentry committed:
fix(dashboards): Wrap customMeasurementsProvider fetchCustomMeasurements in feature flag (#36673)

wraps fetchCustomMeasurements request in feature flag.

Commit:
feat(vitals): vitals alert project data (#36546)

This PR modifies the endpoint for organization web vitals which will be used in an upcoming A/B test:

Adds information about the number of events that have LCP, FCP, cold start, and warm start at an organization level. The UI will use those counts to determine if we have enough data for an organization to warrant showing the notification (min 100 for the relevant category)
Adds project level data for the projects a user has access to. This is needed for organizations who cannot select multiple projects at a time. When we send the user to the web vitals page, we will need to pick a project the user has access to. We can do that by finding the project which has the most relevant transactions for the category we are picking

Commit:
feat(eslint): Disable eslint rule to restrict withRouter imports for existing components (#36629)

wedamija committed:
fix(mep): Make sure we pass organization id when creating query builder (#36685)

Missed passing this in a few places

Fixes SENTRY-VET

barkbarkimashark committed:
fix(release-notification): bypass digest for release notifications  (#36684)

* dont digest for ActiveReleaseAlertNotifications

* cleaner branched logic

asottile-sentry committed:
ref: use actions/setup-python directly (sentry) (#36687)

scttcper committed:
test(ui): Convert range slider to RTL (#36646)

97varun committed:
ref(msteams): Use message builder for help, installation and identity messages (#36495)

Move the JSON blobs for help, installation and identity related messages to a message builder hierarchy like the one we use for Slack. Currently, all the cards that we use for the Microsoft Teams notifications are in the form of JSON blobs in sentry/integrations/msteams/card_builder.py. This is not good for maintainability and there is lot of code duplication.

PixelDust22 committed:
feat(ui): Add multiple checkbox component (#36645)

* feat(ui): Add multiple checkbox component

* Add storybook

* Change text location

asottile-sentry committed:
ref: colorize pytest output in CI by default (#36690)

asottile-sentry committed:
Revert "Revert "ref: fix pip/wheel flipflop by using the pinned versions (#36636)" (#36648)" (#36691)

This reverts commit 096dc6d6c0dc6bbf7be2286700a0a1a63f3c4f3a.

scttcper committed:
test(ui): Convert resolve to RTL (#36676)

wedamija committed:
feat(mep): Add `PerformanceMetricsEntitySubscription` for use with MEP alerts (#36653)

This adds `PerformanceMetricsEntitySubscription` for performance alerts. This allows us to create
performance alerts against either the existing transactions datast, or the new generic metrics
dataset. This isn't in use yet, we'll plug this into the API in a follow-up.

wedamija committed:
tests(mep): Add missing tests for entity subscription helper functions (#36689)

I missed adding tests to `get_entity_subscription_from_snuba_query` and
`get_entity_key_from_snuba_query` when I added them. Including here.

vpandiarajan20 committed:
feat(integrations): Endpoint for SentryFunctions creation page (#36557)

* feat(integrations): Dummy endpoint for SentryFunctions creation page

* fix(integrations): Changed file type, fix url

Changed text file to py file
Changed url for sentry functions backend to be more in line with current style

* fix(integrations): made api private

* feat(integrations): new table

* fix(integrations): debugging in process

* some refactoring

* fix(integrations): add imports to inits

* feat(integrations): Added feature flag and tests for Sentry Functions

* feat(integrations): remove null fill from author element in SentryFunction

* add TODO

* style(lint): Auto commit lint changes

* test(integrations): Add test to make sure the SentryFunction is being created and serialized properly

* style(integrations): remove extraneous comments

* ref(integrations): remove extraneous custom manager

Co-authored-by: Stephen Cefali <scefali@sentry.io>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

vpandiarajan20 committed:
Revert "style(integrations): route family fix & remove redundant code"

This reverts commit 831997e6396f4357923faed4a0b7474ed6bebe12.

vpandiarajan20 committed:
Merge branch 'master' of https://github.com/getsentry/sentry into SentryFunctions(Frontend)

Neo-Zhixing commented:
Do you mind sharing a screenshot for the changes?

vpandiarajan20 committed:
feat(integrations): hid sentry functions with feature flag

vpandiarajan20 commented:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/52717428/179115344-e8421b3a-43a1-467c-b294-5dfffe3e6f95.png">

markstory commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,63 @@
+];
+
+export default function SentryFunctionDetails(props: Props) {
+  const form = useRef(new FormModel());
```
If you don't need to interact with the form model you could remove this as the `Form` component will make its own instance.

vpandiarajan20 replied:
Keeping this around, as I'll need it later

markstory commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,63 @@
+  return (
+    <div>
+      <Feature features={['organizations:sentry-functions']}>
+        <h1>Sentry Function Details</h1>
```
```suggestion
        <h1>{t('Sentry Function Details')}</h1>
```

User visible text should be wrapped in translation functions so that we can add translations more easily in the future.

markstory commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,63 @@
+          model={form.current}
+          onPreSubmit={() => addLoadingMessage(t('Saving changes..'))}
+        >
+          <JsonForm forms={[{title: 'Sentry Function Details', fields: formFields}]} />
```
```suggestion
          <JsonForm forms={[{title: t('Sentry Function Details') fields: formFields}]} />
```

vpandiarajan20 committed:
fix(integrations): making text machine readable

vpandiarajan20 committed:
fix(integrations): small typo

Neo-Zhixing commented on static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx:
```diff
@@ -0,0 +1,63 @@
+        <Form
+          apiMethod={method}
+          apiEndpoint={endpoint}
+          model={form.current}
```
Can you add an error handler `onSubmitError` and `onSubmitSuccess` to clear  and update the loading message with `addErrorMessage` or `addSuccessMessage`?

vpandiarajan20 committed:
feat(integrations): added onSubmitSuccess, onSubmitError

PixelDust22 reviewed (approved):
lgtm
